<launch>
    <arg name="robot_name"/>
    <param name="/use_sim_time" value="true"/>
    <node pkg="gmapping" type="slam_gmapping" name="$(eval arg('robot_name') + '_slam_gmapping')" output="screen">
        
        <remap from="scan" to="$(eval arg('robot_name') + '/base_scan_1')"/>
        
        <param name="map_frame" value="$(eval arg('robot_name') + '/map')"/>
        <param name="base_frame" value="$(eval arg('robot_name') + '/base_link')"/>
        <param name="odom_frame" value="$(eval arg('robot_name') + '/odom')"/>
        <param name="map_update_interval" value="30.0"/>
        <param name="maxUrange" value="4.0"/>
        <param name="sigma" value="0.05"/>
        <param name="kernelSize" value="1"/>
        <param name="lste12p" value="0.05"/>
        <param name="astep" value="0.05"/>
        <param name="iterations" value="5"/>
        <param name="lsigma" value="0.075"/>
        <param name="ogain" value="3.0"/>
        <param name="lskip" value="0"/>
        <param name="srr" value="0.01"/>
        <param name="srt" value="0.02"/>
        <param name="str" value="0.01"/>
        <param name="stt" value="0.02"/>
        <param name="linearUpdate" value="0.5"/>
        <param name="angularUpdate" value="0.436"/>
        <param name="temporalUpdate" value="-1.0"/>
        <param name="resampleThreshold" value="0.5"/>
        <param name="particles" value="80"/>
        <param name="xmin" value="-20.0" />
        <param name="ymin" value="-20.0" />
        <param name="xmax" value="20.0" />
        <param name="ymax" value="20.0" />
        <param name="delta" value="0.025"/>
        <param name="llsamplerange" value="0.01"/>
        <param name="llsamplestep" value="0.01"/>
        <param name="lasamplerange" value="0.005"/>
        <param name="lasamplestep" value="0.005"/>
    </node>
</launch>
